js/webrtc.js

```diff
@@ -581,6 +637,8 @@ var spreedPeerConnectionTable = [];
 
 									videoView.setConnectionStatus(OCA.Talk.Views.VideoView.ConnectionStatus.FAILED_NO_RESTART);
 								}
+							} else {
+								handleIceFailedWithMcu();
```

Comment:
You are forcing a reconnect if either the own stream or one of the peers fails. Did you test what happens when you are in a call with multiple peers and your connection to the MCU gets interrupted? In that scenario, you will receive a `failed` event for all of the streams in a short period and will force multiple reconnects from what I understand.

It probably would be better to start a short timer on a `failed` event to be able to combine multiple events into a single reconnect.

Reply:
@danxuliu :ping_pong: